src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,45 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null");
+        }
+
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login cannot be less than " + MIN_LOGIN_LENGTH);
+        }
+
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null");
+        }
+
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password cannot be less than " + MIN_PASSWORD_LENGTH);
+        }
+
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age cannot be null");
+        }
+
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("Age cannot be less than " + MIN_AGE);
+        }
+
+        if (user.getAge() > Integer.MAX_VALUE) {
+            throw new RegistrationException("Invalid age");
+        }
```

Comment:
Checking if the age is greater than Integer.MAX_VALUE is redundant because `age` is already of type Integer and cannot be higher than Integer.MAX_VALUE by definition. Consider removing this check.